diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30pool mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30pool
--- e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30pool	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30pool	2007-07-09 05:08:06.000000000 -0600
@@ -0,0 +1,8 @@
+{
+    my $pool = $DB->get_prop('ntpd', 'NTPServer');
+
+    if ($pool eq 'pool.ntp.org')
+    {
+      $DB->set_prop('ntpd', 'NTPServer', 'smeserver.pool.ntp.org')
+    }
+}
