diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/MemLimit mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/MemLimit
--- e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/MemLimit	2005-06-23 17:19:22.000000000 -0600
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/MemLimit	2007-01-22 09:39:40.000000000 -0700
@@ -1 +1 @@
-6000000
+12000000
diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/ntpMemLimit mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/ntpMemLimit
--- e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/ntpMemLimit	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/ntpMemLimit	2007-01-22 09:42:03.000000000 -0700
@@ -0,0 +1,5 @@
+{
+    my $ntpd = $DB->get('ntpd');
+    return unless $ntpd;
+    $ntpd->set_prop('MemLimit', 12000000) if $ntpd->prop('MemLimit') < 12000000;
+}
