diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/datetime mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/datetime
--- e-smith-ntp-1.16.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/datetime	2003-04-04 07:40:32.000000000 +1000
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/datetime	2008-02-13 22:29:22.000000000 +1100
@@ -103,10 +103,6 @@
       <trans>NTP Server</trans>
     </entry>
     <entry>
-      <base>SAVE</base>
-      <trans>Save</trans>
-    </entry>
-    <entry>
       <base>VERIFY_DATE_AND_TIME</base>
       <trans>Verify date and time</trans>
     </entry>
@@ -232,11 +228,6 @@
       <base>CURRENT_SETTING</base>
       <trans>Current setting</trans>
     </entry>
-
-    <entry>
-      <base>Configuration</base>
-      <trans>Configuration</trans>
-    </entry>
     <entry>
       <base>Date and time</base>
       <trans>Date and time</trans>
--- e-smith-ntp-1.16.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/datetime.add2general	2008-07-07 19:18:05.000000000 +0200
+++ e-smith-ntp-1.16.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/datetime	2008-07-07 19:19:53.000000000 +0200
@@ -165,10 +165,6 @@
       <trans>Error occurred while setting system time and hardware clock.</trans>
     </entry>
     <entry>
-      <base>ERR_UPDATING_CONF</base>
-      <trans>Error occurred while updating system configuration.</trans>
-    </entry>
-    <entry>
       <base>SERVER_DISABLED</base>
       <trans>Network time server disabled successfully</trans>
     </entry>
