diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/SyncToHWClockSupported mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/SyncToHWClockSupported
--- e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/SyncToHWClockSupported  1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/defaults/ntpd/SyncToHWClockSupported        2006-10-02 15:09:16.678407000 -0400
@@ -0,0 +1 @@
+yes
diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30NTPRunDisconnected mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30NTPRunDisconnected
--- e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30NTPRunDisconnected  1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/db/configuration/migrate/30NTPRunDisconnected        2006-10-02 15:08:28.487522000 -0400
@@ -0,0 +1,10 @@
+{
+    my $ntpd = $DB->get('ntpd');
+    return unless $ntpd;
+    return if $ntpd->get_prop('SyncToHWClockSupported');
+
+    if ($ntpd->get_prop('status') ne 'enabled')
+    {
+        $ntpd->set_prop('status', 'enabled');
+    }
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/etc/e-smith/templates/etc/ntp.conf/00timeServer mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/templates/etc/ntp.conf/00timeServer
--- e-smith-ntp-1.16.0/root/etc/e-smith/templates/etc/ntp.conf/00timeServer 2004-10-19 06:51:14.000000000 -0400
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/etc/e-smith/templates/etc/ntp.conf/00timeServer  2006-10-02 15:08:28.517522000 -0400
@@ -2,13 +2,17 @@
     # vim: ft=perl:
     my $server = $ntpd{NTPServer};
 
-    if ($server =~ /pool.ntp.org/)
+    if ($server !~ /^\s*$/)
     {
-	# The 0, 1, 2 prefixes ensure NTP pool round-robin
-        $OUT .= "server $_.$server\n" for (0..2);
-    }
-    else
-    {
-        $OUT .= "server $server\n";
+
+        if ($server =~ /pool.ntp.org/)
+        {
+                   # The 0, 1, 2 prefixes ensure NTP pool round-robin
+            $OUT .= "server $_.$server\n" for (0..2);
+        }
+        else
+        {
+            $OUT .= "server $server\n";
+        }
     }
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-ntp-1.16.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/datetime.pm mezzanine_patched_e-smith-ntp-1.16.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/datetime.pm
--- e-smith-ntp-1.16.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/datetime.pm        2003-05-09 12:40:44.000000000 -0400
+++ mezzanine_patched_e-smith-ntp-1.16.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/datetime.pm      2006-10-02 15:08:28.547520000 -0400
@@ -156,12 +156,15 @@
     my $now_string = $self->gen_locale_date_string();
 
     my $oldNTPServer = '';
+    my $useNetworkNTPServer = 0;
     $conf = esmith::ConfigDB->open();
     my $rec = $conf->get('ntpd');
     if ($rec)
     {
         $oldNTPServer = $rec->prop('NTPServer') || '';
     }
+
+    $useNetworkNTPServer = 1 if ($oldNTPServer ne '');
     my $ntpEnabled = 0;
 
     if ($rec and $rec->prop('status') eq "enabled")
@@ -188,21 +191,21 @@
                        '12' => $self->localise('DECEMBER'));
 
     # create radio buttons in the same group
-    my ($rbDisableNTP, $rbEnableNTP) = 
-        $q->radio_group(-name => 'ntpStatus', 
+    my ($rbDisableNetworkNTP, $rbEnableNetworkNTP) = 
+        $q->radio_group(-name => 'networkNtpStatus', 
             -values => ['disabled', 'enabled'],
-            -default => $ntpEnabled ? 'enabled' : 'disabled',
+            -default => $useNetworkNTPServer ? 'enabled' : 'disabled',
             -linebreak => 'true', 
             -labels => {enabled => '', disabled => ''});
 
     print "<tr><td colspan=2><table>\n";
-
-    if (! $ntpEnabled) 
+    print '<input type="hidden" name="ntpStatus" value="' . ($ntpEnabled ? "enabled" : "disabled") . '">';
+    if (! $useNetworkNTPServer) 
     {
 
 
       print $q->Tr(
-        $q->td({-class => "sme-radiobutton"}, $rbDisableNTP),
+        $q->td({-class => "sme-radiobutton"}, $rbDisableNetworkNTP),
         $q->td($q->h3($self->localise('SET_DATE_TITLE')))),
       $q->Tr(
         $q->td({-class => "sme-radiobutton"}, "&nbsp;"),
@@ -264,7 +267,7 @@
     my $ntpEnabledDesc;
     my $showTime;
 
-    if ($ntpEnabled)
+    if ($useNetworkNTPServer)
     {
       $ntpEnabledTitle = $self->localise('NTP_CONFIGURE_TITLE');
       $ntpEnabledDesc = $self->localise('NTP_CONFIGURE_DESC');
@@ -278,7 +281,7 @@
     }
 
     print $q->Tr(
-      $q->td({-class => "sme-radiobutton"}, $rbEnableNTP),
+      $q->td({-class => "sme-radiobutton"}, $rbEnableNetworkNTP),
       $q->td($q->h3($ntpEnabledTitle))),
     $q->Tr(
       $q->td({-class => "sme-radiobutton"}, "&nbsp;"),
@@ -300,10 +303,10 @@
       )
     ),"\n";
 
-    if ($ntpEnabled)
+    if ($useNetworkNTPServer)
     {
       print $q->Tr(
-        $q->td({-class => "sme-radiobutton"}, $rbDisableNTP),
+        $q->td({-class => "sme-radiobutton"}, $rbDisableNetworkNTP),
         $q->td($q->h3 ($self->localise("NTP_DISABLE_TITLE")))),
       $q->Tr(
         $q->td({-class => "sme-radiobutton"}, "&nbsp;"),
@@ -351,9 +354,11 @@
     my $newNtpServer = $q->param('ntpServer') || ''; 
     my $newNtpStatus = $q->param('ntpStatus') || '';
 
-    # If ntp status was and still is disabled, then set the time.
-    if (($oldNtpStatus eq 'disabled') &&
-        ($newNtpStatus eq 'disabled'))
+    # If ntp status was and still is disabled, or if the ntp server was and 
+    # still is not set then set the time.
+
+    if ((($oldNtpStatus eq 'disabled') && ($newNtpStatus eq 'disabled')) ||
+        (($oldNtpServer eq '') && ($newNtpServer eq '')))
     {   
         $self->_performSetDateTime();
     }
@@ -710,14 +715,16 @@
         $msg .= $q->p($self->localise('SERVER_DISABLED_DESC'));
 
     }
-    else    # enable service and synch with ntpServer
+    else    # enable service
     {   
         if ($ntpServer =~ /^([a-zA-Z0-9\.\-]+)$/)
         {
+            # sync with network time server
             $ntpServer = $1;
         }
         elsif ($ntpServer =~ /^\s*$/)
         {
+            # sync with rtc
             $ntpServer = "";
         }
         else

